reactor-core/src/main/java/reactor/core/publisher/FluxBufferTimeout.java

```diff
@@ -162,6 +168,21 @@ void nextCallback(T value) {
 					values = v;
 				}
 				v.add(value);
+
+				long next;
+				long o = outstanding;
```

Comment:
if I understand correctly, the intent here is to accommodate the fact that there would be more `onNext` than `outstanding`? but down below `outstanding` is updated to reflect the request made upstream, so this cannot happen (or rather, if it does then we have a non-compliant source). Can you produce a test that triggers `o <= 0L` path?

You could alternatively simplify this though?
```java
if (OUTSTANDING.decrementAndGet(this) <= 0) {
    throw Exceptions.failWithOverflow("unrequested element received");
}
```

Reply:
You're right - that logic is a lot simpler! I have implemented that change with a couple of differences:

- The check is on the new value of outstanding being less than 0, since no items outsanding is a valid state
- Instead of directly throwing, actual.onError is called with the exception
- I moved the check to before the code adding the item to the buffer, since there's no point in adding to the buffer if an error is being returned.

Doing this actually revealed an issue with the timing of the increase to OUTSTANDING. I had been increasing it after the request to upstream for more items. This led to the possibility that the item would be delivered before OUTSTANDING increased, which would trigger the error. I have moved the request to upstream to after the increase in OUTSTANDING